### PR TITLE
Block metadata

### PR DESCRIPTION
**Pull request by robdmoore**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,6 +43,11 @@
       "import": "./index.mjs",
       "require": "./index.js"
     },
+    "./block": {
```
[the remark this answers is not in the document]
Comment (robdmoore) [reply package.json:46]: Deliberately moved it, it's undocumented so while it's a breaking change I don't mind.

I've resolved the weird type and exported transform now.

File: src/transform.ts [hunk 2]
```diff
@@ -537,3 +537,235 @@ function mapKeys<TKey>(map: Map<TKey, any>): TKey[] {
...
+function countAllTransactions(txns: (BlockTransaction | BlockInnerTransaction)[]): number {
+  return txns.reduce((sum, txn) => sum + 1 + countAllTransactions(txn.dt?.itx ?? []), 0)
+}
+
+/**
+ * Extracts balance changes from a block transaction or inner transaction.
+ * @param transaction The transaction
+ * @returns The balance changes
+ */
+export function extractBalanceChangesFromBlock(transaction: BlockTransaction | BlockInnerTransaction): BalanceChange[] {
```
Comment (neilcampbell) [inline src/transform.ts:570]:
```suggestion
export function extractBalanceChangesFromBlockTransaction(transaction: BlockTransaction | BlockInnerTransaction): BalanceChange[] {
```

**Review by neilcampbell — APPROVED** [review]
Nice!
I also noticed that main current has a peer dependency on a beta version of utils-ts. Assuming we are good to release 6.0.0 of utils-ts and update this to a non beta version?

**Comment by robdmoore** [comment]
There's a bit of work to do on utils first before releasing v6 ideally. There might be a couple of breaking changes I'm tossing up.

**Pull request merged**